app/assets/javascripts/mobile/mobile.js

```diff
@@ -125,57 +125,83 @@ $(document).ready(function(){
     }
   });
 
-  /* Show comments */
-  $("a.show_comments", ".stream").bind("tap click", function(evt){
-    evt.preventDefault();
-    var link = $(this),
+  var showComments = function(element, show_form) {
+    var link = $(element),
         parent = link.closest(".bottom_bar").first(),
+        activeElements = parent.find('.comment_action, .show_comments'),
         commentsContainer = function(){ return parent.find(".comment_container").first(); },
         existingCommentsContainer = commentsContainer();
 
-    if( link.hasClass('active') ) {
-      existingCommentsContainer.hide();
-      if(!link.hasClass('bottom_collapse')){
-        link.removeClass('active');
+    var initializeReactionsView = function() {
+      // Update 'reactions' link and 'pen' button to reflect the current state
+      activeElements.removeClass('inactive').addClass('active');
```

Comment:
I'm wondering why we need two class here, only one and using toggle should be enough. This would imply to also refactor the css though so if you don't want to touch it, that's fine.